drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -372,16 +404,91 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
+multilane:
+		dev_dbg(bus->dev, "%s: could not find clock value for bandwidth %d, checking multi-lane\n",
 			__func__, bus->params.bandwidth);
-		return -EINVAL;
+
+		m_rt = list_last_entry(&bus->m_rt_list, struct sdw_master_runtime, bus_node);
+		/*
+		 * Find available manager lanes that connected to the first Peripheral. No need
+		 * to check all Peripherals available lanes because we can't use multi-lane
+		 * if we can't find any available lane for the first Peripheral.
+		 */
+		s_rt = list_first_entry(&m_rt->slave_rt_list, struct sdw_slave_runtime, m_rt_node);
+		slave_prop = &s_rt->slave->prop;
+
+		for (i = 1; i < SDW_MAX_LANES; i++) {
+			if (!slave_prop->lane_maps[i])
+				continue;
+
+			required_bandwidth = 0;
+			list_for_each_entry(m_p_rt, &m_rt->port_list, port_node) {
+				required_bandwidth += m_rt->stream->params.rate *
+					hweight32(m_p_rt->ch_mask) *
+					m_rt->stream->params.bps;
+			}
+			if (required_bandwidth <= curr_dr_freq - bus->lane_used_bandwidth[i]) {
+				/* Check if m_lane is connected to all Peripherals */
+				if (!is_all_peripherals_connected(m_rt, slave_prop->lane_maps[i])) {
+					dev_dbg(bus->dev,
+						"some Peripherals are not connected to %d\n",
+						slave_prop->lane_maps[i]);
+					continue;
+				}
+				m_lane = slave_prop->lane_maps[i];
+				dev_dbg(&s_rt->slave->dev,
+					"M lane %d P lane %d can be used\n",
+					m_lane, i);
+				bus->lane_used_bandwidth[i] += required_bandwidth;
+				/*
+				 * Use non-zero manager lane, subtract the lane 0
+				 * bandwidth that is already calculated
+				 */
+				bus->params.bandwidth -= required_bandwidth;
+				use_multi_lane = true;
+				break;
+			}
+		}
+
+		if (!use_multi_lane) {
+			dev_err(bus->dev, "%s: could not find an available lane\n", __func__);
+			return -EINVAL;
+		}
+
+		/* Set Peripheral lanes */
+		list_for_each_entry(s_rt, &m_rt->slave_rt_list, m_rt_node) {
+			slave_prop = &s_rt->slave->prop;
+			for (i = 1; i < SDW_MAX_LANES; i++) {
+				if (slave_prop->lane_maps[i] == m_lane) {
+					dev_dbg(&s_rt->slave->dev, "Set Peripheral lane = %d\n", i);
+					list_for_each_entry(s_p_rt, &s_rt->port_list, port_node) {
+						s_p_rt->lane = i;
+						break;
+					}
+				}
+			}
+		}
+		/*
+		 * Set Manager lanes. Configure the last m_rt in bus->m_rt_list only since
+		 * we don't wan't to touch other m_rts that are already working.
+		 */
+		list_for_each_entry(m_p_rt, &m_rt->port_list, port_node) {
+			m_p_rt->lane = m_lane;
+		}
 	}
 
 	ret = sdw_select_row_col(bus, curr_dr_freq);
 	if (ret < 0) {
-		dev_err(bus->dev, "%s: could not find frame configuration for bus dr_freq %d\n",
-			__func__, curr_dr_freq);
-		return -EINVAL;
+		if (use_multi_lane) {
+			dev_err(bus->dev,
+				"%s: could not find frame configuration for bus dr_freq %d\n",
+				__func__, curr_dr_freq);
+			return -EINVAL;
+		}
+		dev_warn(bus->dev,
+			 "%s: could not find frame configuration for bus dr_freq %d try multi-lane\n",
+			 __func__, curr_dr_freq);
```

Comment:
I don't think this should be a dev_warn since we know it's a legit configuration supported on some commercial devices.

Reply:
> I don't think this should be a dev_warn since we know it's a legit configuration supported on some commercial devices.

Changed it to dev_dbg